common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/NativeAuthOAuth2Strategy.kt

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.identity.common.java.providers.nativeauth
+
+import androidx.annotation.VisibleForTesting
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Strategy
+import com.microsoft.identity.common.java.providers.oauth2.OAuth2StrategyParameters
+
+/**
+ * The implementation of native authentication API OAuth2 client.
+ */
+class NativeAuthOAuth2Strategy(
+    private val strategyParameters: OAuth2StrategyParameters,
+    @get:VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    val config: NativeAuthOAuth2Configuration
+) :
+    MicrosoftStsOAuth2Strategy(config, strategyParameters) {
+    private val TAG = NativeAuthOAuth2Strategy::class.java.simpleName
+
+    override fun getIssuerCacheIdentifierFromTokenEndpoint(): String {
+        if (config.useRealAuthority) {
+            return super.getIssuerCacheIdentifierFromTokenEndpoint()
+        } else {
+            return "login.windows.net"
```

Comment:
I'm sure this would be defined as a const somewhere already..let's use that. If not then let's define it rather than hard coding strings in multiple places.